includes/data-stores/class-wc-admin-reports-coupons-data-store.php

```diff
@@ -345,4 +346,22 @@ public static function sync_order_coupons( $order_id ) {
 		}
 	}
 
+	/**
+	 * Clean coupons data when an order is deleted.
+	 *
+	 * @param int $order_id Order ID.
+	 */
+	public static function sync_on_order_delete( $order_id ) {
```

Comment:
Since this function is the same for all data stores, should we move it into `WC_Admin_Reports_Data_Store` to dry things up a bit?

Reply:
Since we need new hooks for https://github.com/woocommerce/wc-admin/pull/1459 probably will make no big difference.